packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -226,6 +221,25 @@ const properties = {};
   );
 });
 
+// defaultValue stringifies boolean values for backward compatability
+// from when it was treated as a reserved property. Reserved
+// properties skip most property validation.
+properties.defaultValue = new PropertyInfoRecord(
```

Comment:
Hope this is clearer.

Reply:
How long has it been "treated as a reserved property"? If it's a few minors then we can treat this as a bug we're fixing.

Reply:
Man... this takes me back. I _think_ they were added as an exception to the rules when the warning was first created:

https://github.com/facebook/react/commit/de1de9e18f8faf153ad673c8206a2918e3e81035#diff-24dece27bd4b2d550d6d7cf69604586cR27

Man. We've been doing this a long time 😨

Reply:
Looks like React 15.1.0

Reply:
Doing a quick local test with the fixtures, I can confirm this behavior as far back as 0.14.0

Reply:
If we ignore the differences in warnings, is there any *difference in behavior* we are taking about? If so, how long has the difference in behavior been there?

Reply:
As far as I know, the only differences in behavior are for replacing Symbols and Functions with empty strings when when we added custom attributes in 16.

As long as I can remember, `defaultValue` and `value` were "anything goes". Any value gets stringified or turned into an empty string on `null/undefined`.

Later today, I'll be able to do some archaeology.

Reply:
OK. My suggestion is: it's fine to change warnings even if old behavior was there for a few majors (but was wrong). It's fine to change actual *behavior* if it was buggy and was only there for a few minors. Or if there were no legit use cases for relying on it.